internal/api/create_credentials_v1alpha1.go

```diff
@@ -34,34 +34,63 @@ func (s *server) CreateCredentials(
 		return nil, connect.NewError(connect.CodeUnimplemented, errSecretManagementDisabled)
 	}
 
-	creds := credentials{
-		project:        req.Msg.GetProject(),
-		name:           req.Msg.GetName(),
-		description:    req.Msg.GetDescription(),
-		credType:       req.Msg.GetType(),
-		repoURL:        req.Msg.GetRepoUrl(),
-		repoURLIsRegex: req.Msg.GetRepoUrlIsRegex(),
-		username:       req.Msg.GetUsername(),
-		password:       req.Msg.GetPassword(),
+	credType := req.Msg.GetType()
+
+	if err := validateFieldNotEmpty("type", credType); err != nil {
+		return nil, err
 	}
 
+	switch credType {
+	case kargoapi.CredentialTypeLabelValueGit,
+		kargoapi.CredentialTypeLabelValueHelm,
+		kargoapi.CredentialTypeLabelValueImage:
+		creds := specificCredentials{
+			project:        req.Msg.GetProject(),
+			name:           req.Msg.GetName(),
+			description:    req.Msg.GetDescription(),
+			credType:       req.Msg.GetType(),
+			repoURL:        req.Msg.GetRepoUrl(),
+			repoURLIsRegex: req.Msg.GetRepoUrlIsRegex(),
+			username:       req.Msg.GetUsername(),
+			password:       req.Msg.GetPassword(),
+		}
+
+		kubernetesSecret, err := s.createSpecificCredentials(ctx, creds)
+
+		if err != nil {
+			return nil, err
+		}
+
+		return connect.NewResponse(
+			&svcv1alpha1.CreateCredentialsResponse{
+				Credentials: kubernetesSecret,
+			},
+		), nil
+	}
+
+	return nil, connect.NewError(
+		connect.CodeInvalidArgument,
+		errors.New("type should be one of git, helm, image"),
+	)
+}
+
+// creates credentials used for known purpose; specifically external subscriptions
+// to private git repo or helm OCI or docker image
+func (s *server) createSpecificCredentials(ctx context.Context, creds specificCredentials) (*corev1.Secret, error) {
```

Comment:
I think linters might only enforce it for exported things, but the generally accepted practice in Go docs is that a comment like this one should start with the name of the variable, function, method, etc. that it is documenting. It seems somewhat silly, but iirc many doc tools in the Go ecosystem assume this convention is observed.

```suggestion
// createSpecificCredentials creates credentials used specifically for
// git, Helm chart, and image container repositories.
func (s *server) createSpecificCredentials(ctx context.Context, creds specificCredentials) (*corev1.Secret, error) {
```